Setup JPA

MrZhenShen opened the pull request:
**Change type: #minor**
[Learn more](https://semver.org)

# Description
Delete ddl-auto update from configuration

## Create entities:
* Game
* GameSession
* MoneyBalance
* Result
* Round
* User
## Create enums for entities:
* Role
* ResultType
## Create repositorires:
* GameRepository
* GameSessionRepository
* MoneyBalanceRepository
* ResultRepository
* UserRepository

MrZhenShen committed:
remove ddl-auto from config
create entities and repositories

MrZhenShen committed:
fix excludes for jacoco

nadiiamalytska9 commented on src/main/java/sigma/internship/petProject/entity/Game.java:
```diff
@@ -0,0 +1,46 @@
+@Table(name = "game")
+@Getter
+@Setter
+@ToString
```
It's better to use shortcut for annotations ToString, EqualsAndHashCode, Getter on all fields, Setter on all non-final fields, and RequiredArgsConstructor - `@Data` from a Lombok.
For more information: https://projectlombok.org/features/Data

Please, use `@Data` annotation in all corresponding cases.

nadiiamalytska9 commented on src/main/java/sigma/internship/petProject/entity/Game.java:
```diff
@@ -0,0 +1,46 @@
+@Setter
+@ToString
+@NoArgsConstructor
+public class Game {
```
I suggest you to add one more annotation for all your entities classes - `@Builder`. 
It would help you to create and set data easily, using just builder.

nadiiamalytska9 commented on src/main/java/sigma/internship/petProject/entity/Game.java:
```diff
@@ -0,0 +1,46 @@
+import lombok.Setter;
+import lombok.ToString;
+
+import javax.persistence.*;
```
It is recommended to not use `.*` wildcard imports. Several reasons from stackoverflow:
- Because it puts a lot of stuff into your namespace (might shadow some other object from previous import and you won't know about it).
- Because you don't know exactly what is imported and can't easily find from which module a certain thing was imported (readability).

Specify only specific imports, which you use in your project.

nadiiamalytska9 commented on src/main/java/sigma/internship/petProject/entity/Game.java:
```diff
@@ -0,0 +1,46 @@
+    public void createEntity() {
+        createDate = LocalDateTime.now();
+    }
+}
```
Please, add a blank line at the end of every code file.
Here you can find why it is necessary: 
https://stackoverflow.com/questions/2402593/whats-the-reason-for-leaving-an-extra-blank-line-at-the-end-of-a-code-file

MrZhenShen replied:
In PR it is shown only added/removed lines, while in IDEA there is everything with these blank lines, but in any case I will take it into consideration next time

nadiiamalytska9 replied:
Yes, you are right, I double checked, everything is okay

nadiiamalytska9 commented on src/main/java/sigma/internship/petProject/entity/Game.java:
```diff
@@ -0,0 +1,46 @@
+    private boolean isOneUserAction;
+
+    @Column(name = "winning", nullable = false)
+    private double winning;
```
It's better use `BigDecimal` type for all values, connected to 'money', instead of `float` and `double`. It's a good practice to use `BigDecimal` to control roundings with a good precision. 
But more information you can read, for example, here https://stackoverflow.com/questions/6320209/javawhy-should-we-use-bigdecimal-instead-of-double-in-the-real-world

You can also easily set up precision and scale for you 'money' values, for example:
```
@Column(name = "winning", precision = 19, scale = 2, nullable = false)
    private BigDecimal winning;
```

MrZhenShen committed:
use @Data and @Builder annotations instead some specific
import specific components instead of '.*'
for money variables set BigDecimal data type

nadiiamalytska9 reviewed (approved):
looks good to me 
but next time, please, add answers to your review comment, if you added corresponding changes or not. Its's a good practice, so a reviewers can see, that changes were added.
